services/assets.ts

```diff
@@ -0,0 +1,51 @@
+import * as Options from '../options';
+import { BaseService } from '../infrastructure';
+import { Asset, AssetData, ListAssetData } from '../models';
+
+/**
+ * A service for manipulating a Shopify shop's theme asset.
+ */
+export class Assets extends BaseService {
+    constructor(shopDomain: string, accessToken: string) {
+        super(shopDomain, accessToken, "themes");
+    }
+
+    /**
+     * Retrieves a single asset for a theme
+     * @param id Id of the theme.
+     * @param key he path to the asset within a theme.
+     * @param options Options for filtering the result.
+     */
+    public get(id: number, key: string, options?: Options.FieldOptions) {
+        return this.createRequest<AssetData>("GET", `${id}/assets.json?asset[key]=${key}&theme_id=${id}`, "", options);
+    }
+
+    /**
+     * Creates or updates an asset for a theme.
+     * You can include the `src` or `source_key` property to create the asset from an existing file.
+     * @param id Id of the assets being updated.
+     * @param assets The updated asset.
+     */
+    public update(id: number, asset: Asset) {
+        return this.createRequest<AssetData>("PUT", `${id}/assets.json`, "", { asset });
+    }
+
+    /**
+     * Retrieves a list of assets for a theme
+     * @param id Id of the theme.
+     * @param options Options for filtering the results.
+     */
+    public list(id: number, options?: Options.FieldOptions) {
+        return this.createRequest<ListAssetData>("GET", `${id}/assets.json`, "", options);
```

Comment:
```suggestion
        return this.createRequest<Asset[]>("GET", `${id}/assets.json`, "assets", options);
```

Reply:
The api response is different than others shopify api responses.

**The response of get a single asset is:**
```
{
  asset: ... as Asset
}
```
`asset` is not plural 

**The response of get assets.json is:**

```
{
  assets: ... as Asset[]
}
```
`assets` is plural.

**The request is not just**
```
[
    {...} as Asset,
    {...} as Asset,
    ...
]
```
or
```
{...} as Asset
```
So it would be consistent with the rest of api, but unfortunately it is not.

But if you update an asset 

on the `get` request but it is `assets` on the `list` request.

That is the reason why I have additionally created the Data interfaces.

Reply:
Ah I see what you're saying! However, the `createRequest` function should be able to handle that without the need for `AssetData` or `ListAssetData` interfaces. `createRequest` uses that third string to parse out a property from JSON and return it directly, so if the API is returning `{"asset": {...}}` you should be able to use `createRequest("GET", url, "asset", option)` and the function will return that asset prop. The data goes from `{ "asset": {asset props}}` to `{asset props}`.

Reply:
Okay thank you, will will update this